app/main.py

```diff
@@ -1,3 +1,23 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cache_hash_result = {}
+
+    @wraps(func)
+    def results_save_wrapper(*args, **kwargs) -> None:
+        func_arg_kwarg_hash = hash(args) + hash(func)
+
+        for item in kwargs.items():
+            func_arg_kwarg_hash += hash(item)
```

Comment:
why do u need to hash all values ? try to simplify ur code make a key like this f.e. :
func_arg_kwarg_hash = (*args, *kwargs.values(), hash(func)) no need to make a loop for the same functionlity

Reply:
> why do u need to hash all values ? try to simplify ur code make a key like this f.e. : func_arg_kwarg_hash = (*args, *kwargs.values(), hash(func)) no need to make a loop for the same functionlity

btw this case was kinda what i was wandering about:
![image](https://user-images.githubusercontent.com/107141441/216713214-ef4111d0-6ebf-4a44-99a1-de4248472c77.png)